components/playground/instance/tiflash.go

```diff
@@ -131,26 +132,37 @@ func (inst *TiFlashInstance) Start(ctx context.Context, version utils.Version) e
 
 	args := []string{
 		"server",
-	}
-	args = append(args,
 		fmt.Sprintf("--config-file=%s", configPath),
 		"--",
-		fmt.Sprintf("--tmp_path=%s", filepath.Join(inst.Dir, "tmp")),
-		fmt.Sprintf("--path=%s", filepath.Join(inst.Dir, "data")),
-		fmt.Sprintf("--listen_host=%s", inst.Host),
-		fmt.Sprintf("--logger.log=%s", inst.LogFile()),
-		fmt.Sprintf("--logger.errorlog=%s", filepath.Join(inst.Dir, "tiflash_error.log")),
-		fmt.Sprintf("--status.metrics_port=%d", inst.StatusPort),
-		fmt.Sprintf("--flash.service_addr=%s", utils.JoinHostPort(AdvertiseHost(inst.Host), inst.ServicePort)),
-		fmt.Sprintf("--raft.pd_addr=%s", strings.Join(endpoints, ",")),
-		fmt.Sprintf("--flash.proxy.addr=%s", utils.JoinHostPort(inst.Host, inst.ProxyPort)),
-		fmt.Sprintf("--flash.proxy.advertise-addr=%s", utils.JoinHostPort(AdvertiseHost(inst.Host), inst.ProxyPort)),
-		fmt.Sprintf("--flash.proxy.status-addr=%s", utils.JoinHostPort(inst.Host, inst.ProxyStatusPort)),
-		fmt.Sprintf("--flash.proxy.data-dir=%s", filepath.Join(inst.Dir, "proxy_data")),
-		fmt.Sprintf("--flash.proxy.log-file=%s", filepath.Join(inst.Dir, "tiflash_tikv.log")),
-	)
-
-	var err error
+	}
+	runtimeConfig := [][]string{
+		{"tmp_path", filepath.Join(inst.Dir, "tmp")},
+		{"path", filepath.Join(inst.Dir, "data")},
+		{"listen_host", inst.Host},
+		{"logger.log", inst.LogFile()},
+		{"logger.level", "debug"},
+		{"logger.errorlog", filepath.Join(inst.Dir, "tiflash_error.log")},
+		{"status.metrics_port", fmt.Sprintf("%d", inst.StatusPort)},
+		{"flash.service_addr", utils.JoinHostPort(AdvertiseHost(inst.Host), inst.ServicePort)},
+		{"raft.pd_addr", strings.Join(endpoints, ",")},
+		{"flash.proxy.addr", utils.JoinHostPort(inst.Host, inst.ProxyPort)},
+		{"flash.proxy.advertise-addr", utils.JoinHostPort(AdvertiseHost(inst.Host), inst.ProxyPort)},
+		{"flash.proxy.status-addr", utils.JoinHostPort(inst.Host, inst.ProxyStatusPort)},
+		{"flash.proxy.data-dir", filepath.Join(inst.Dir, "proxy_data")},
+		{"flash.proxy.log-file", filepath.Join(inst.Dir, "tiflash_tikv.log")},
+	}
+	userConfig, err := unmarshalConfig(configPath)
+	if err != nil {
+		return errors.Trace(err)
+	}
```

Comment:
we should continue the following logic when the config file in `configPath` is not exists

Reply:
```go
	configPath := filepath.Join(inst.Dir, "tiflash.toml")
	if err := prepareConfig(
		configPath,
		inst.ConfigPath,
		inst.getConfig(),
	); err != nil {
		return err
	}
```

`configPath` always exists.

Reply:
got it